com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/archive/publisher/ZipPublisher.java

```diff
@@ -145,6 +145,7 @@ public void publish(ArchiveEntry entry, byte[] data) throws CompileExceptionErro
         final String archiveEntryHexdigest = entry.getHexDigest();
         final String archiveEntryName = entry.getName();
         final String zipEntryName = (archiveEntryHexdigest != null) ? archiveEntryHexdigest : archiveEntryName;
+        getEntries().put(archiveEntryName, entry);
```

Comment:
Could we perhaps cache `getEntries()` in the constructor? Or make `entries` in publisher `protected`?